dataservice/api/participant/models.py

```diff
@@ -14,3 +23,7 @@ class Participant(db.Model, Base):
     """
     __tablename__ = "participant"
     external_id = db.Column(db.String(32))
+    diagnoses = db.relationship('Diagnosis', secondary=diagnoses_assoc_table,
```

Comment:
**Using lazy=joined for Participant.diagnoses**
Chose joined load for Participant.diagnoses to support the MVP use case: get single Participant by id. We decided this query would result in a response that includes the Person’s full demographic info, full diagnoses info, and full samples info. Thus it makes sense to use joined/subquery load parameter here so that we can retrieve all of this information for a person in a single query. 

**Choice of lazy=joined or lazy=subquery**
In terms of choosing joined vs subquery, several sources say that in general (i.e. in most use cases) join is more performant than subquery. However, it seems that for Postgres this should not matter since it is able to intelligently choose whether or not to use a subquery or convert the subquery to a join. Thus, I’ve decided to use lazy=joined rather than subquery. This can always change later based on the evolution of our use cases.

Reply:
I agree that we'll probably want to fetch all the diagnoses with each participant for many use cases, however, what if there is one case where we just want to get all Partipant's ids, and we forget to set `lazy=True`? Then we've joined and sent all the diagnosis data unnecessarily.

Reply:
Isn't that the same thing as forgetting to use lazy=joined or joinedload in our get endpoints?
Maybe we could have the default behavior of our get endpoints use lazy=joined/subquery (for demographic, diagnosis, sample) and then once we start supporting partial object retrieval we can explicitly set lazy=True for get all queries. 

I agree though probably just easier to figure it out once we can play with data

Reply:
As for the postgres comment, I had done a quick search and took a glance at this article and this video. 

https://robots.thoughtbot.com/postgresql-performance-considerations
https://www.youtube.com/watch?v=dAQhBJMAeqc

Reply:
Forgetting to do `lazy=joined` is still equally as possible, but the difference is that executing a statement with `lazy=joined` that doesn't need to be joined is much more costly than one that is lazy.

Reply:
👍